src/flutter-pi.c

```diff
@@ -1810,25 +1816,63 @@ static int init_application(void) {
     FlutterEngineResult engine_result;
     FlutterProjectArgs project_args = {0};
     void *libflutter_engine_handle;
+    char *libflutter_engine_path;
     int ok;
 
+    asprintf(&libflutter_engine_path, "%s/libflutter_engine.so", flutterpi.flutter.asset_bundle_path);
+
     libflutter_engine_handle = NULL;
+
+    libflutter_engine_handle = dlopen(libflutter_engine_path, RTLD_LOCAL | RTLD_NOW);
+
+    if (libflutter_engine_handle == NULL)
+    {   
+
+        
+        printf("[flutter-pi] Warning: Could not load libflutter_engine.so inside the asset bundle : Trying to open libflutter_engine.so.");
+        printf(
+            flutterpi.flutter.runtime_mode == kDebug ? "debug" :
+            flutterpi.flutter.runtime_mode == kProfile ? "profile" :
+            "release"
+        );
+        printf("\n");
+        fprintf(
+            stderr,
+            dlerror()
+        );
+        printf("\n");
+    }
+
+    free(libflutter_engine_path);
+
+    if (libflutter_engine_handle == NULL)
+    {
     if (flutterpi.flutter.runtime_mode == kRelease) {
         libflutter_engine_handle = dlopen("libflutter_engine.so.release", RTLD_LOCAL | RTLD_NOW);
         if (libflutter_engine_handle == NULL) {
-            LOG_ERROR("[flutter-pi] Warning: Could not load libflutter_engine.so.release: %s. Trying to open libflutter_engine.so...\n", dlerror());
+            LOG_ERROR("Warning: Could not load libflutter_engine.so.release: %s. Trying to open libflutter_engine.so.debug", dlerror());
+            printf("\n");
+        }
+    } else if (flutterpi.flutter.runtime_mode == kProfile) {
+        libflutter_engine_handle = dlopen("libflutter_engine.so.profile", RTLD_LOCAL | RTLD_NOW);
+        if (libflutter_engine_handle == NULL) {
+            LOG_ERROR("Warning: Could not load libflutter_engine.so.profile: %s Trying to open libflutter_engine.so.debug", dlerror());
+            printf("\n");
         }
     } else if (flutterpi.flutter.runtime_mode == kDebug) {
         libflutter_engine_handle = dlopen("libflutter_engine.so.debug", RTLD_LOCAL | RTLD_NOW);
         if (libflutter_engine_handle == NULL) {
-            LOG_ERROR("[flutter-pi] Warning: Could not load libflutter_engine.so.debug: %s. Trying to open libflutter_engine.so...\n", dlerror());
+            LOG_ERROR("Warning: Could not load libflutter_engine.so.debug: %s Trying to open libflutter_engine.so ", dlerror());
+            printf("\n");
```

Comment:
Why did you put the `\n` into a separate printf here and above?
The printf will go to stdout, LOG_ERROR will go to stderr